src/libraries/Directory.Build.targets

```diff
@@ -285,4 +285,15 @@
     </When>
   </Choose>
 
+  <!-- The upfront vertical builds already generates the dependency file. -->
+  <PropertyGroup Condition="'$(BuildingInsideVisualStudio)' == 'true' and Exists('$(ProjectDepsFilePath)')">
+    <GenerateDependencyFile>false</GenerateDependencyFile>
+  </PropertyGroup>
+
+  <Target Name="CheckVSRestoreRequirements"
+          Condition="'$(BuildingInsideVisualStudio)' == 'true'"
+          BeforeTargets="ResolvePackageAssets" >
+    <Error Text="Please run dotnet restore from command line for $(MSBuildProjectName) because the project contains platform specific target frameworks." Condition="!Exists('$(ProjectAssetsFile)')" />
```

Comment:
This error text doesn't match the condition.  It's saying the project has platform specific TFMs, yet the condition is only checking if the assets file is missing.

Reply:
Is that sufficient? Don't you also need to disable NuGet restore in the Nuget restore settings inside VS?

Reply:
Right, that's required to prevent the package manager error (for the cases where it would fail) and then you probably want an error like this that tells folks to run from the commandline.

Reply:
> Is that sufficient? Don't you also need to disable NuGet restore in the Nuget restore settings inside VS?

i don't want to do that because that will disable the restore for all the tests that are not platform specific.

Reply:
Can you put a condition on the property which disables restore?

Reply:
we use a nuget.config file to disable restore for the sln and i was not able find to exclude the nuget.config file from certain projects. 
We can always add the nuget.config to the individual projects rather than libraries root but that would be a lot of duplication.